tensorflow/compiler/mlir/xla/ir/hlo_ops.td

```diff
@@ -648,6 +648,27 @@ def HLO_CompareOp: HLO_Op<"compare",
   >];
 }
 
+def HLO_UniqueCountOp: HLO_UnaryElementwiseOp<"unique_count",
+      [NoSideEffect], HLO_IntTensor> {
+    let builders = [OpBuilder<
+    "Builder *builder, OperationState &result, Value operand"
+    >];
+}
+
+def HLO_UniqueOp: HLO_Op<"unique",
+      [NoSideEffect]> {
+  let arguments = (ins
+    HLO_Tensor:$x,
+    HLO_IntTensor:$count
+  );
+  let builders = [
+    OpBuilder<"Builder *builder, OperationState &result, Value left, Value right">,
+  ];
+  let results = (outs
```

Comment:
HLOs only have 1 output, this is how XLA works and this dialect matches XLA HLO client side.

Reply:
> HLOs only have 1 output, this is how XLA works and this dialect matches XLA HLO client side.

You mean that we should separate the two outputs in different ops? like:

def HLO_UniqueOp1: HLO_Op<"unique1",
      [NoSideEffect]> {
   // ...
   let results = (outs HLO_Tensor:$y)
}

def HLO_UniqueOp2: HLO_Op<"unique2",
      [NoSideEffect]> {
   // ...
   let results = (outs HLO_Tensor:$idx)
}

Reply:
> XLA would use tuple for multiple result right? MLIR does not have this restriction though, we should look closer into how much artificial restrictions we carry over.
> 
> The more interesting question I'd have is: what is this `unique` op end up being if we try to export to XLA HLOs?

Actually, XLA not support `unique` now, we want to compiler embedding graph via mlir which xla can not do this.
So maybe we should not export to XLA HLO now?

Reply:
Yes, so we currently have this dialect closely matching XLA HLO restrictions/behavior. We are working on updating some parts around it to make it easier and to make it clearer where the restrictions are artificial ...

Reply:
> Yes, so we currently have this dialect closely matching XLA HLO restrictions/behavior. We are working on updating some parts around it to make it easier and to make it clearer where the restrictions are artificial ...

You means that hlo dialect will support multi-output in the near future? I think there are many other multi-output ops can be compiled via MLIR. 
Another way is that lowering these ops which not through HLO/LHLO.

Reply:
Correct that is the other approach. And as to HLO dialect supporting mulit-output in near future, that is an open discussion point and I think one where we'd want to do that in a way that doesn't complicate the current interop too much (e.g., inserting operations to make and deconstruct tuples during export).